pcsx2/Patch.h

```diff
@@ -69,6 +69,9 @@ namespace Patch
 		std::string name;
 		std::string description;
 		std::string author;
+		std::string group;
```

Comment:
Trees are part of the UI, not the patch system itself. This should not be in here - see how it's implemented in the Qt interface.

Reply:
removed

Reply:
from my understanding an research. Qt makes use of helper functions in this struct to parse the name. Which says that trees are in fact part of the patch system , since the nodes are effectively parsed via the name

Reply:
It's really not - the structure is not intended to be used directly in the UI, the core function abstracts the parsing of the file away from the UI, so it can be shared between interfaces, but how the data is presented is entirely UI side. That's why Qt doesn't do anything with `PatchInfo` after the widgets are set up.

Reply:
> the structure is not intended to be used directly in the UI. 

PatchInfo structure is passed as a parameter ? The FullscreenUI I was modifying directly interacts with the patchinfo structure.

Reply:
Indeed it does. But it also doesn't display the hierarchy ;) 

It's a "quick and dirty" way of exposing the cheat/patch list, because otherwise you wouldn't be able to toggle them via a gamepad at all. Better than nothing.

Reply:
> Indeed it does. But it also doesn't display the hierarchy ;)

I don't really follow that logic. There is no explicit inheritance or hierarchy defined in regards to the PatchInfo struct.
https://github.com/PCSX2/pcsx2/blob/a34e139843dcff8aa3d6cc5a5e856ae706920849/pcsx2/Patch.h#L67-L75

Reply:
Exactly. The assigning of items to locations in the tree is done in https://github.com/PCSX2/pcsx2/blob/master/pcsx2-qt/Settings/GameCheatSettingsWidget.cpp - it uses GetNamePart() to set the label for the widget, and GetNameParentPart() to determine which node to attach it to (which may be further subdivided, based on the number of nesting levels).

Reply:
You mentioned "But it also doesn't display the hierarchy ;)"

What hiearchy was there to display ? There is no friend class or inheritance to speak of?

> it uses GetNamePart()
right back to this. So instead of maintaining the variable you utilize a function to parse the string (in a UI class accessing PatchInfo) The whole point of not maintaining the variable is so that PatchInfo is not accessed directly by the ui ... so effectively helper functions were made to parse the data (an unknown amount of times) 

I don't follow the logic

Reply:
It's the hierarchy in terms of the patches. A\B\C, for example.

Read the Qt implementation. It's fairly easy to follow, recursion is used to set up multiple levels of nesting.